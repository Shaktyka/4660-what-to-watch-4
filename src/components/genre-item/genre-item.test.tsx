import React from 'react';
import renderer from 'react-test-renderer';

import GenreItem from './genre-item.jsx';

describe(`GenreItem rendering`, () => {

  it(`GenreItem renders correctly like clicked element`, () => {
    const genre = `Comedy`;

    const tree = renderer
      .create(
          <GenreItem
            genre={genre}
            isActive={true}
            onGenreClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`GenreItem renders correctly`, () => {
    const genre = `Comedy`;

    const tree = renderer
      .create(
          <GenreItem
            genre={genre}
            isActive={false}
            onGenreClick={() => {}}
          />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
