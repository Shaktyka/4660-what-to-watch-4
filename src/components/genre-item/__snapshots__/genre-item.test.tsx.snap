// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreItem rendering GenreItem renders correctly 1`] = `
<li
  className="catalog__genres-item "
  onClick={[Function]}
>
  <a
    className="catalog__genres-link"
    href="#"
  >
    Comedy
  </a>
</li>
`;

exports[`GenreItem rendering GenreItem renders correctly like clicked element 1`] = `
<li
  className="catalog__genres-item catalog__genres-item--active"
  onClick={[Function]}
>
  <a
    className="catalog__genres-link"
    href="#"
  >
    Comedy
  </a>
</li>
`;
